plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/FacetUninstallDelegate.java

```diff
@@ -95,4 +94,53 @@ protected IStatus run(IProgressMonitor monitor) {
     uninstallJob.schedule();
 
   }
+
+  private void removeAppEngineJarsFromMavenProject(IProject project, IProgressMonitor monitor) throws CoreException {
+    IMavenProjectFacade facade = MavenPlugin.getMavenProjectRegistry().getProject(project);
+    Model pom = facade.getMavenProject(monitor).getModel();
+
+    List<Dependency> currentDependencies = pom.getDependencies();
+    List<Dependency> updatedDependencies = updateMavenProjectDependencies(currentDependencies);
+    pom.setDependencies(updatedDependencies);
+
+    Properties properties = pom.getProperties();
+    updatePomProperties(properties, monitor);
+
+    DefaultModelWriter writer = new DefaultModelWriter();
+    File pomFile = pom.getPomFile();
+    try {
+      writer.write(pomFile, null /* options */, pom);
+    } catch (IOException e) {
+      throw new CoreException(StatusUtil.error(FacetUninstallDelegate.class, e.getMessage()));
+    }
+    project.getFile(pomFile.getName()).refreshLocal(IResource.DEPTH_ZERO, monitor);
+  }
+
+  //visible for testing
+  public static List<Dependency> updateMavenProjectDependencies(List<Dependency> initialDependencies) {
+    List<Dependency> finalDependencies = new ArrayList<Dependency>();
+    if ((initialDependencies == null) || (initialDependencies.isEmpty())) {
+      return finalDependencies;
+    }
+
+    List<Dependency> allAppEngineDependencies = MavenAppEngineFacetUtil.getAppEngineDependencies();
+    for (Dependency dependency : initialDependencies) {
+      if(!MavenAppEngineFacetUtil.doesListContainDependency(allAppEngineDependencies, dependency)) {
+        finalDependencies.add(dependency);
+      }
+    }
+
+    return finalDependencies;
+  }
+
+  //visible for testing
+  public static void updatePomProperties(Properties projectProperties, IProgressMonitor monitor) {
```

Comment:
https://github.com/GoogleCloudPlatform/cloud-tools-for-eclipse/pull/627/files#r77026534